### ISSUE TITLE
Sweep_: format wandb run name in pytorch checkpoint

### ISSUE THREAD

**Issue by ayulockin** [ayulockin/pytorch-lightning#5]
### Description & Motivation

It would be great to format the wandb run name: 
`wandb.run.name`

into the checkpoint filename.

epoch: 10
loss: 0.5423

filename: "epoch{epoch}-loss{val_loss:.2f}" --> "epoch10-loss0.54"

filename: "{wandb.run.name}-{epoch}-{val_loss:.2f}" --> "swept_sky_54-epoch10-loss0.54"

I am not sure if this is already possible but I think the filename function in the ModelCallback can only access metrics so far. 
 # Suggested Solution: 
 This is a great suggestion. However, currently, PyTorch Lightning does not directly support using the wandb run name in the checkpoint name. You have to handle this manually in your code. Let's discuss a possible workaround.

Here, we will override the `ModelCheckpoint`'s `__init__` and `_format_checkpoint_name` methods:

```python
from pytorch_lightning import Callback
from pytorch_lightning.callbacks import ModelCheckpoint
from pytorch_lightning.utilities import rank_zero_warn
from typing import Dict, Union

class MyModelCheckpoint(ModelCheckpoint):

    def __init__(self, wandb, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.wandb = wandb

    def _format_checkpoint_name(
        self, filename: Union[str, None], metrics: Dict[str, float], ver: Union[int, None]
    ) -> str:
        if not self._check_monitor(metrics):
            return
        filename = super()._format_checkpoint_name(filename, metrics, ver)

        filename = (
            filename.replace("{epoch}", metrics["epoch"] or "").replace("{step}", metrics["global_step"] or "")
        )
  
        filename = filename.replace("{wandb.run.name}", self.wandb.run.name or "")
   
        try:
            # lightning adds these now
            filename = filename.replace("={val_loss:.2f}", "=0.54").replace("={loss:.2f}", "=0.56")
        except KeyError:
            rank_zero_warn(
                f'Key(s) in filename "{self.filename}" is/are not present in the'
                " monitor's records. Replaced with blank.",
                RuntimeWarning,
            )
        return self.CHECKPOINT_JOIN_CHAR.join([self.dirpath, filename]) if self.dirpath else filename
```

This will replace `{wandb.run.name}` in the filename with the actual wandb run name.

Keep in mind that this is just a workaround and its efficiency heavily depends on your specific use-case. It is recommended to test the updated code extensively before deploying. 

Please feel free to suggest this as a new feature on the official PyTorch Lightning GitHub page. The team always appreciates feedback and suggested improvements from users.
